### ISSUE TITLE
Add a dry-run option

### ISSUE THREAD

**Issue by magicDGS** [robmaz/distmap#58]
Currently, the `--verbose` option prints the parsed inputs in the terminal and then exit; nevertheless, there is no real check for the arguments or requirements (such as ReadTools version or existance of files).

It will be nice to debug the current behaviour to have a `--dry-run` option that will construct the pipeline internally and do the checks, printing on the command line what the pipeline will be (this can be a good addition also if not dry-run is specified, at the beginning of the output). For example, it could say:

```
PIPELINE (job = ${job_description})
===================================
* Indexing (${reference})
* Upload (${inputs})
* Trimming (${trimming_script} with arguments: "${trim_args}")
* Map (${mapper} with arguments  "${mapper_args}")

WARNING: trimming is perform on upload and it is OBSOLETE
```

This will be useful for debugging, but also for the final user to check how the procedure will be.

**Comment by robmaz** [comment]
Good idea in principle, but difficult to implement with the current convoluted structure.

**Comment by magicDGS** [comment]
So for the long term!
